src/main/java/com/cag/cagbackendapi/service/validation/ValidationService.java

```diff
@@ -0,0 +1,32 @@
+package com.cag.cagbackendapi.service.validation;
+
+import com.cag.cagbackendapi.constants.DetailedErrorMessages;
+import com.cag.cagbackendapi.error.exceptions.UnauthorizedException;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ValidationService {
+
+    @Value("${spring.authKey}")
+    private String authKey;
+
+    private final Logger logger;
+
+    public ValidationService(Logger logger) {
+        this.logger = logger;
+    }
+
+    public void validateAuthKey(String requestAuthKey) throws UnauthorizedException {
+        if (requestAuthKey == null || requestAuthKey.isEmpty()) {
+            logger.error(DetailedErrorMessages.MISSING_AUTH_KEY);
+            throw new UnauthorizedException(DetailedErrorMessages.MISSING_AUTH_KEY, null);
+        }
+
+        if (!requestAuthKey.equals(authKey)) {
+            logger.error(DetailedErrorMessages.WRONG_AUTH_KEY);
+            throw new UnauthorizedException(DetailedErrorMessages.WRONG_AUTH_KEY, null);
```

Comment:
What does the null value do in this case?

Reply:
I made a class called UnauthorizedException that takes a String "message" and a Throwable "cause". Custom exceptions are used to hide the real ugly errors and display to our users a clean and nice error. For example, if we tried to divide by zero somewhere in our code - Java would throw some "IllegalOperationException" with an ugly message and a bunch of details our users won't need or understand. Using custom exceptions like the one that I made above will take the ugly error message and give the user instead "Unauthorized Exception: The auth key was incorrect.".

Reply:
Sorry, I missed the initial question. The reason that "cause" is null is because I'm throwing the exception myself so I created the exception. In the future we will have try/catch blocks where we catch errors. Inside of a catch block, we will have caught an error. At that point we could throw our own custom error with the "cause" set to the actual exception we caught in our catch block (Different from the custom exception we manually threw)